src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/SubjectAccessRequestService.kt

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Goal
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.ActionPlanService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.Induction
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.InductionNotFoundException
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.service.InductionService
+import uk.gov.justice.hmpps.kotlin.sar.HmppsPrisonSubjectAccessRequestService
+import uk.gov.justice.hmpps.kotlin.sar.HmppsSubjectAccessRequestContent
+import java.time.LocalDate
+import java.time.ZoneOffset
+
+@Service
+class SubjectAccessRequestService(
+  private val inductionService: InductionService,
+  private val actionPlanService: ActionPlanService,
+) : HmppsPrisonSubjectAccessRequestService {
+  data class SubjectAccessRequestContent(val induction: Induction?, val goals: List<Goal>)
```

Comment:
I think this data class is wrong ...
The `SubjectAccessRequestContent` is a REST API response type, and you have it correctly defined in the swagger spec, complete with the API model versions of the Induction and Goals.
Because it's in the swagger spec you'll find it gets generated for you, so you dont need to write the class again here.

But the bigger problem is that they actually contain different types.

The generated class is an API model class, and contains properties of induction and goals which are also API model classes. This is correct.

But your definition here is not an API model type (its in a different package), and crucially the induction and goal properties are NOT API model types. The types here are the domain classes, which we never surface through the API. The domain classes may contain fields that we would not want to surface through the REST API.

There are mappers that map between the domain classes and the corresponding REST API classes

Reply:
I think this more like what you want the service to do, making use of the REST API model classes, and the mappers to map from domain to model :

```
package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service

import org.springframework.stereotype.Service
import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.resource.mapper.actionplan.GoalResourceMapper
import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.resource.mapper.induction.InductionResourceMapper
import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.ActionPlanService
import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.InductionNotFoundException
import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.service.InductionService
import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.SubjectAccessRequestContent
import uk.gov.justice.hmpps.kotlin.sar.HmppsPrisonSubjectAccessRequestService
import uk.gov.justice.hmpps.kotlin.sar.HmppsSubjectAccessRequestContent
import java.time.LocalDate
import java.time.ZoneOffset

@Service
class SubjectAccessRequestService(
  private val inductionService: InductionService,
  private val actionPlanService: ActionPlanService,
  private val inductionResourceMapper: InductionResourceMapper,
  private val goalResourceMapper: GoalResourceMapper,
) : HmppsPrisonSubjectAccessRequestService {

  override fun getPrisonContentFor(
    prn: String,
    fromDate: LocalDate?,
    toDate: LocalDate?,
  ): HmppsSubjectAccessRequestContent? {
    val fromDateInstance = fromDate?.atStartOfDay()?.toInstant(ZoneOffset.UTC)
    val toDateInstance = toDate?.atStartOfDay()?.toInstant(ZoneOffset.UTC)

    val goals = actionPlanService.getActionPlan(prn).goals
      .filter { fromDateInstance == null || it.createdAt?.isAfter(fromDateInstance) ?: true }
      .filter { toDateInstance == null || it.createdAt?.isBefore(toDateInstance) ?: true }
      .map {
        goalResourceMapper.fromDomainToModel(it)
      }
      .toSet()

    val induction = try {
      inductionService.getInductionForPrisoner(prn)
        .takeIf { fromDateInstance == null || it.createdAt?.isAfter(fromDateInstance) ?: true }
        .takeIf { toDateInstance == null || it?.createdAt?.isBefore(toDateInstance) ?: true }
        ?.let {
          inductionResourceMapper.toInductionResponse(it)
        }
    } catch (e: InductionNotFoundException) {
      null
    }

    if (goals.isEmpty() && induction == null) return null
    return HmppsSubjectAccessRequestContent(
      content = SubjectAccessRequestContent(induction, goals),
    )
  }
}
```